### DISCUSSION TITLE
provide an option to use BGP with or without MetalLB

### DISCUSSION THREAD

**Discussion by AskAlice** [onedr0p/cluster-template#686] [General]
Given you use BGP on your personal cluster, the benefit should be obvious, but ripping apart your configure script so that it will work without metallb has been some work, but if you could make an optional configuration where the speaker is disabled.
ie: https://metallb.universe.tf/configuration/calico/

**Comment by onedr0p** [comment]
I would love to do something like this but I'm pretty reluctant to add more logic in the bash script, it's not maintainable to add so many conditions and optional features.

Ideally that whole configure script and env should be refactored into a Go / Python script that can read from a config file but I haven't had the time or ambition to take this on 🫠

**Comment by onedr0p** [comment]
Now that the template is using Cilium this will also be possible using the BGPCP with Cilium in v1.14. I am still not feeling great about adding the logic to support this in the configure scripts current state.

Also brought up here: https://github.com/onedr0p/flux-cluster-template/issues/769

**Reply by AskAlice** [reply]
makes perfect sense. The configure script has in fact turned into spaghetti. spaghetti everywhere.

nonetheless I appreciate you dropping me an update and I'm glad you're trying out a different solution that could encompass this

**Comment by snoopy82481** [comment]
You can use BGPCP in 1.13.x, I currently have that setup in my cluster working with out issue.

You just need to provide a CiliumBGPPeeringPolicy and CiliumLoadBalancerIPPool
